pkg/operator/rules.go

```diff
@@ -166,7 +168,7 @@ func ValidateRule(promRuleSpec monitoringv1.PrometheusRuleSpec) []error {
 
 // Select selects PrometheusRules and translates them into native Prometheus/Thanos configurations.
 // The second returned value is the number of rejected PrometheusRule objects.
-func (prs *PrometheusRuleSelector) Select(namespaces []string) (map[string]string, int, error) {
+func (prs *PrometheusRuleSelector) Select(recorder record.EventRecorder, namespaces []string) (map[string]string, int, error) {
```

Comment:
(nit) maybe rather a field in the PrometheusRuleSelector struct?

Reply:
Good catch, let's make it consistent with ResourceSelector :)